pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -3208,28 +3210,30 @@ PJ_DEF(pjsua_acc_id) pjsua_acc_find_for_incoming(pjsip_rx_data *rdata)
         return PJSUA_INVALID_ID;
     }
 
-    uri = rdata->msg_info.to->uri;
+    to_header_uri = rdata->msg_info.to->uri;
+    request_uri = rdata->msg_info.msg->line.req.uri;
 
     PJSUA_LOCK();
 
     /* Use Req URI if To URI is not SIP */
-    if (!PJSIP_URI_SCHEME_IS_SIP(uri) &&
```

Comment:
This and the following if-statements that include PJSIP_URI_SCHEME_IS_SIP(S) could be improved. I don't know why 
```cpp
if (rdata->msg_info.msg->type == PJSIP_REQUEST_MSG)
```
exists, so I haven't touched it.